### ISSUE TITLE
`ZonedDateTime` for times before the epoch do not serialize correctly 

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-modules-java8#69]
(moved from https://github.com/FasterXML/jackson-datatype-jsr310/issues/97
reported by @twilson-palantir )

Version 2.8.10
```
    public static void main(String[] args) throws Exception {
        ObjectMapper mapper = new ObjectMapper().registerModule(new JavaTimeModule());
        ZonedDateTime original = ZonedDateTime.parse("1959-04-13T05:05:38.599Z");
        String serialized = mapper.writeValueAsString(original);
        ZonedDateTime deserialized = mapper.readValue(serialized, ZonedDateTime.class);
        System.out.println(original);
        System.out.println(serialized);
        System.out.println(deserialized);
    }
```
outputs
```
1959-04-13T05:05:38.599Z
-338324062.599000000
1959-04-13T05:05:37.401Z[UTC]
```
I haven't checked other types for issues, but I wouldn't be surprised if they were similarly broken.

**Comment by cowtowncoder** [comment]
Not actually sure this is "easy" per se, but I think that investigating basic problem should be straight-forward and then it should be possible to either:

1. Add unit test to show the problem (or, if can not reproduce, close as fixed)
2. Possibly fix, OR, if more difficult than expected, add findings

and if not easily fixable, add a failing unit test (under `failing` package so it won't block builds)

**Comment by cowtowncoder** [comment]
Fixed via #132, duplicates #120 -- will be in 2.10.0.pr2.
